packages/jest-resolve/src/__tests__/resolve.test.ts

```diff
@@ -345,13 +345,66 @@ describe('findNodeModule', () => {
       );
     });
 
+    test('supports nested pattern', () => {
+      const result = Resolver.findNodeModule('#nested', {
+        basedir: path.resolve(importsRoot, './nested-import/index.cjs'),
+        conditions: ['node', 'require'],
+      });
+
+      expect(result).toEqual(
+        path.resolve(importsRoot, './nested-import/node.cjs'),
+      );
+    });
+
+    describe('supports array pattern', () => {
+      test('resolve to first found', () => {
+        const result = Resolver.findNodeModule('#array-import', {
+          basedir: path.resolve(importsRoot, './array-import/index.cjs'),
+          conditions: ['import'],
+        });
+
+        expect(result).toEqual(
+          path.resolve(importsRoot, './array-import/node.mjs'),
+        );
+      });
+
+      test('skip over not met nested condition', () => {
+        const result = Resolver.findNodeModule('#array-import', {
+          basedir: path.resolve(importsRoot, './array-import/index.cjs'),
+          conditions: ['browser'],
+        });
+
+        expect(result).toEqual(
+          path.resolve(importsRoot, './array-import/browser.cjs'),
+        );
+      });
+
+      test('match nested condition', () => {
+        const result = Resolver.findNodeModule('#array-import', {
+          basedir: path.resolve(importsRoot, './array-import/index.cjs'),
+          conditions: ['chrome', 'browser'],
+        });
+
+        expect(result).toEqual(
+          path.resolve(importsRoot, './array-import/chrome.cjs'),
+        );
+      });
+    });
+
     test('fails for non-existent mapping', () => {
       expect(() => {
         Resolver.findNodeModule('#something-else', {
           basedir: path.resolve(importsRoot, './foo-import/index.js'),
           conditions: [],
         });
-      }).toThrow('Missing "#something-else" import in "foo-import" package');
+      }).toThrow(
+        expect.objectContaining({
+          code: 'ERR_PACKAGE_IMPORT_NOT_DEFINED',
+          message: expect.stringMatching(
+            /^Package import specifier "#something-else" is not defined in package/,
```

Comment:
@unional bug in the message here.

`Package import specifier \"#something-else\" is not defined in package /Users/simen/repos/jest/packages/jest-resolve/src/__mocks__/imports/foo-importpackage.json imported from /Users/simen/repos/jest/packages/jest-resolve/src/__mocks__/imports/foo-import/index.js`

`foo-importpackage.json` is missing a slash